### PR TITLE
skip assign_wcs if the MSAMETFL keyword is not populated correctly

### LINKED ISSUE #2107: Make opening of MSAMETFL more robust in assign_wcs
An update was recently made to the `asn_generator` rules so that level-2b ASN's are not created if an NRS_MSASPEC exposure does not have an `MSAMETFL` (MSA metadata file) specified, which means level-2b processing won't get triggered within the Ops environment.

But it's still possible for someone to try to manually execute `calwebb_spec2` on an NRS_MSASPEC exposure that doesn't have `MSAMETFL` defined or is set to "N/A". This causes an error in the `assign_wcs.nirspec.get_msa_metadata` function. The logic in that function should be "beefed up" to check for conditions like "MSAMETFL = 'N/A'", issue an error message to the log, and shutdown cleanly, rather than crashing.

### PR DESCRIPTION
**Pull request by nden**
Fixes #2107

### CODE REVIEW

File: jwst/assign_wcs/assign_wcs_step.py [hunk 1]
```diff
@@ -63,10 +63,16 @@ def process(self, input):
...
-            if msa_metadata_file is not None:
-                msa_metadata_file = self.make_input_path(msa_metadata_file)
-                reference_file_names['msametafile'] = msa_metadata_file
+            if input_model.meta.exposure.type == "NRS_MSASPEC":
+                msa_metadata_file = input_model.meta.instrument.msa_metadata_file
+                if msa_metadata_file is not None and msa_metadata_file.strip() not in ["", "N/A"]:
+                    msa_metadata_file = self.make_input_path(msa_metadata_file)
+                    reference_file_names['msametafile'] = msa_metadata_file
+                else:
+                    log.info("MSA metadata file (MSAMETFL) is required for NRS_MSASPEC exposures.")
```
Comment (hbushouse) [inline jwst/assign_wcs/assign_wcs_step.py]: Might want to bump this up to warning level, if not error.

File: jwst/assign_wcs/assign_wcs_step.py [hunk 2]
```diff
@@ -63,10 +63,16 @@ def process(self, input):
...
+                    input_model.meta.cal_step.assign_wcs = 'SKIPPED'
+                    log.warning("assign_wcs: SKIPPED")
+                    return input_model
```
Comment (hbushouse) [inline jwst/assign_wcs/assign_wcs_step.py]: If this occurs, then none of the rest of the `calwebb_spec2` steps can proceed either. Is there already something implemented in `calwebb_spec2` that can detect that `assign_wcs` has been skipped and issue another warning (or error) and shutdown cleanly?

Comment (stscieisenhamer) [reply jwst/assign_wcs/assign_wcs_step.py]: Yep: https://github.com/STScI-JWST/jwst/blob/master/jwst/pipeline/calwebb_spec2.py#L169

File: jwst/assign_wcs/assign_wcs_step.py (hunk 1 above)
Comment (nden) [reply jwst/assign_wcs/assign_wcs_step.py]: done - it's log.error now

**Comment by hbushouse** [comment]
It occurs to me - after the fact of course - that we might also want imbed https://github.com/STScI-JWST/jwst/blob/master/jwst/assign_wcs/nirspec.py#L431 in a `try` block and again shutdown cleanly with an error message if the file can't be opened (e.g. doesn't exist, is in another directory, etc.).

**Comment by nden** [comment]
I'll do that, though to get there one has to call the function directly, in which case they better be giving a name. But it's good to have this check.

**Comment by nden** [comment]
Fixed in #2126

**Pull request merged**